lib/ocpp/v201/profile.cpp

```diff
@@ -448,19 +449,19 @@ ChargingSchedulePeriod minimize_charging_schedule_period_by_limit(
 
     if (prevailing_period.limit == NO_LIMIT_SPECIFIED && candidate_period.limit != NO_LIMIT_SPECIFIED) {
         adjusted_period = candidate_period;
-    } else if (candidate_period.limit != NO_LIMIT_SPECIFIED) {
+    } else if (candidate_period.limit.has_value()) {
         const auto charge_point_max_phases = candidate_period.numberPhases.value_or(default_number_phases);
 
         const auto period_max_phases = prevailing_period.numberPhases.value_or(default_number_phases);
         adjusted_period.numberPhases = std::min(charge_point_max_phases, period_max_phases);
 
         if (current_charging_rate_unit_enum == ChargingRateUnitEnum::A) {
-            if (candidate_period.limit < prevailing_period.limit) {
-                adjusted_period.limit = candidate_period.limit;
+            if (candidate_period.limit.value() < prevailing_period.limit.value_or(DEFAULT_LIMIT_AMPS)) {
+                adjusted_period.limit = candidate_period.limit.value();
             }
         } else {
-            const auto charge_point_limit_per_phase = candidate_period.limit / charge_point_max_phases;
-            const auto period_limit_per_phase = prevailing_period.limit / period_max_phases;
+            const auto charge_point_limit_per_phase = candidate_period.limit.value() / charge_point_max_phases;
+            const auto period_limit_per_phase = prevailing_period.limit.value_or(0) / period_max_phases; // FIXME
 
```

Comment:
TODO: find out implications for backwards compatibility with 2.0.1
(limit became optional)